src/components/video.js

```diff
@@ -16,46 +17,79 @@ const sampleData = {
   spotify: 'https://open.spotify.com/track/4uLU6hMCjMI75M1A2tKUQC',
   likes: 27,
   genres: ['soul', 'pop'],
-  queue: [
-    'zReP_EYZGEw',
-    'dQw4w9WgXcQ',
-    'PJWemSzExXs',
-    'k4V3Mo61fJM',
-    'e-ORhEE9VVg',
-    '2hlT8CqZ2pA',
-    'U0V1y2p1sgs',
-  ],
+  queue: [],
 };
 
 export default function Video(props) {
-  let preQueue;
+  let preQueue = useSelector((state) => state.queue);
   if (props.queue) {
     preQueue = props.queue;
-  } else {
-    preQueue = sampleData.queue;
   }
   let [queue, changeQueue] = useState([
     props.id || sampleData.yid,
     ...preQueue,
   ]);
 
+  let [currentIndex, changeIndex] = useState(0);
+
   useEffect(() => {
     if (document !== undefined) {
       document.body.scrollTop = 0; // For Safari
       document.documentElement.scrollTop = 0; // For Chrome, Firefox, IE and Opera
     }
   }, []);
 
+  let changeURLid = (id) => {
+    if (window !== undefined) {
+      if (window.history.pushState) {
+        var newurl =
+          window.location.protocol +
+          '//' +
+          window.location.host +
+          window.location.pathname +
+          '?=' +
+          id;
+        window.history.pushState({ path: newurl }, '', newurl);
+      }
+    }
+  };
+
+  let playNextVideo = () => {
+    if (queue.length > 0 && currentIndex + 1 < queue.length) {
+      changeURLid(queue[currentIndex + 1]);
+      changeIndex(currentIndex + 1);
+      if (document !== undefined) {
+        let el = document.querySelector('.previous.queueCard');
```

Comment:
Why not add a `.next` and scroll that into view? Currently, the scroll into view is a bit erratic & unpredictable when clicking the previous & next buttons.

Reply:
I have added a better scroll into view function than the default one, you can test the changes now.